opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/AppResources.java

```diff
@@ -71,31 +80,51 @@ public class AppResources
 	@RolesAllowed({AuthorizationCheck.ODCS_API_GUEST})
 	public Response getAppRefs() throws DbException
 	{
-		LOGGER.trace("Getting App Refs.");
-		try (DbInterface dbi = new DbInterface();
-			ApiAppDAO dao = new ApiAppDAO(dbi))
+
+		try (LoadingAppDAI dai = getLegacyDatabase().makeLoadingAppDAO())
+		{
+			List<ApiAppRef> ret = dai.listComputationApps(false)
+					.stream()
+					.map(AppResources::map)
+					.collect(Collectors.toList());
+			return Response.status(HttpServletResponse.SC_OK)
+					.entity(ret).build();
+		}
+		catch (DbIoException ex)
 		{
-			ArrayList<ApiAppRef> ret = dao.getAppRefs();
-			LOGGER.trace("Returning {} apps.", ret.size());
-			return ApiHttpUtil.createResponse(ret);
+			throw new DbException("Unable to retrieve apps", ex);
 		}
 	}
 
+	static ApiAppRef map(CompAppInfo app)
+	{
+		ApiAppRef ret = new ApiAppRef();
+		ret.setAppId(app.getAppId().getValue());
+		ret.setAppName(app.getAppName());
+		ret.setAppType(app.getAppType());
+		ret.setComment(app.getComment());
+		ret.setLastModified(app.getLastModified());
+		return ret;
+	}
+
 	@GET
 	@Path("app")
 	@Produces(MediaType.APPLICATION_JSON)
 	@RolesAllowed({AuthorizationCheck.ODCS_API_GUEST})
- 	public Response getApp(@QueryParam("appid") Long appId)
-		throws WebAppException, DbException, SQLException
+	public Response getApp(@QueryParam("appid") Long appId)
+			throws WebAppException, DbException
 	{
 		if (appId == null)
-			throw new WebAppException(ErrorCodes.MISSING_ID, 
-				"Missing required appid parameter.");
-		LOGGER.debug("Getting app with id {}", appId);
-		try (DbInterface dbi = new DbInterface();
-			ApiAppDAO dao = new ApiAppDAO(dbi))
+			throw new WebAppException(HttpServletResponse.SC_BAD_REQUEST,
+					"Missing required appid parameter.");
+		try (LoadingAppDAI dai = getLegacyDatabase().makeLoadingAppDAO())
+		{
+			return Response.status(HttpServletResponse.SC_OK)
+					.entity(map(dai.getComputationApp(DbKey.createDbKey(appId)))).build();
+		}
+		catch(NoSuchObjectException | DbIoException ex)
```

Comment:
`NoSuchObjectException` should return a 404, while other exceptions should return 500